src/pal/src/misc/dbgmsg.cpp

```diff
@@ -470,66 +470,88 @@ Notes :
 int DBG_printf(DBG_CHANNEL_ID channel, DBG_LEVEL_ID level, BOOL bHeader,
                LPCSTR function, LPCSTR file, INT line, LPCSTR format, ...)
 {
-    CHAR *buffer = (CHAR*)alloca(DBG_BUFFER_SIZE);
-    CHAR indent[MAX_NESTING+1];
-    LPSTR buffer_ptr;
-    INT output_size;
-    va_list args;
-    void *thread_id;
-    int old_errno = 0;
-
-    old_errno = errno;
+    struct ErrnoHolder
+    {
+        int value;
+        ErrnoHolder() : value(errno) { }
+        ~ErrnoHolder()
+        {
+            if ( value != errno )
+            {
+                fprintf( stderr,"ERROR : errno changed from %d to %d inside DBG_printf\n", errno, value );
```

Comment:
I actually wonder if we should remove this error message. It seems like it doesn't help anything to print that. We preserve the errno, so even if the code below modified it, it doesn't matter.